config/http/notifications/streaming-http/http.json

```diff
@@ -0,0 +1,83 @@
+{
+  "@context": "https://linkedsoftwaredependencies.org/bundles/npm/@solid/community-server/^7.0.0/components/context.jsonld",
+  "@graph": [
+    {
+      "comment": "Path prefix used by streaming HTTP receiveFrom endpoints",
+      "@id": "urn:solid-server:default:variable:streamingHTTPReceiveFromPrefix",
+      "valueRaw": ".notifications/StreamingHTTPChannel2023/"
+    },
+    {
+      "comment": "Creates updatesViaStreamingHTTP2023 Link relations",
+      "@id": "urn:solid-server:default:StreamingHTTPMetadataWriter",
+      "@type": "StreamingHTTPMetadataWriter",
+      "baseUrl": { "@id": "urn:solid-server:default:variable:baseUrl" },
+      "pathPrefix": { "@id": "urn:solid-server:default:variable:streamingHTTPReceiveFromPrefix" }
+    },
+    {
+      "comment": "[Override] Allows discovery of the corresponding streaming HTTP channel",
+      "@id": "urn:solid-server:default:MetadataWriter",
+      "@type": "ParallelHandler",
+      "handlers": [
+        { "@id": "urn:solid-server:default:StreamingHTTPMetadataWriter" }
+      ]
+    },
+    {
+      "comment": "Handles the request targeting a StreamingHTTPChannel2023 receiveFrom endpoint.",
+      "@id": "urn:solid-server:default:StreamingHTTP2023Router",
+      "@type": "OperationRouterHandler",
+      "baseUrl": { "@id": "urn:solid-server:default:variable:baseUrl" },
+      "allowedMethods": [ "GET" ],
+      "allowedPathNames": [ "/StreamingHTTPChannel2023/" ],
+      "handler": {
+        "@id": "urn:solid-server:default:StreamingHTTP2023Handler",
+        "@type": "StreamingHTTPRequestHandler",
+        "streamMap": { "@id": "urn:solid-server:default:StreamingHTTPMap" },
+        "pathPrefix": { "@id": "urn:solid-server:default:variable:streamingHTTPReceiveFromPrefix" },
+        "credentialsExtractor": { "@id": "urn:solid-server:default:CredentialsExtractor" },
+        "permissionReader": { "@id": "urn:solid-server:default:PermissionReader" },
+        "authorizer": { "@id": "urn:solid-server:default:Authorizer" }
+      }
+    },
+    {
+      "comment": "[Override] add the router to notification type handler",
+      "@id": "urn:solid-server:default:NotificationTypeHandler",
+      "@type": "WaterfallHandler",
+      "handlers": [
+        { "@id": "urn:solid-server:default:StreamingHTTP2023Router" }
+      ]
+    },
```

Comment:
@joachimvh We were trying to replace the above with 
```jsonld
    {
      "comment": "[Override] add the router to base HTTP handler",
      "@type": "OverrideListInsertBefore",
      "overrideInstance": { "@id": "urn:solid-server:default:BaseHttpHandler" },
      "overrideParameter": { "@id": "WaterfallHandler:_handlers" },
      "overrideTarget": "urn:solid-server:default:LdpHandler",
      "overrideValue": "urn:solid-server:default:StreamingHTTP2023Router"
    },
```
but ended up with this error
```
Invalid predicate IRI: overrideParameter
```

We tried to follow https://componentsjs.readthedocs.io/en/latest/configuration/configurations/overrides/#overridelistinsertbefore

Reply:
The reason is that this feature was added in Components.js v5.5.0 and the CSS still has v5.4.2 in its dependencies.

But the configurations of the main CSS repo should never use overrides though, those were designed for external applications that can't modify the configurations directly. If something is being added in the main repo the necessary changes can just be made to the config. Having overrides in the main repo would make it more difficult for external components to make changes.

All this to say that the new entry should just be added to the array directly.

Reply:
If I add it directly, wouldn't it make StreamingHTTPChannel2023 obligatory? I assumed that all the channel types need to be easy to include or exclude in the final config used by the deployment. By the way, after this PR is ready, will I need to make another one for https://github.com/CommunitySolidServer/configuration-generator?

Reply:
Usually we fix this by having the identifier linked to a handler that does nothing when the relevant config is not included. E.g., when OIDC is disabled this is what the handler linked to its identifier looks like: https://github.com/CommunitySolidServer/CommunitySolidServer/blob/f73dfb31c0fe132524323acf6c4f4636bcd8bc80/config/identity/oidc/disabled.json#L5-L7

But thinking on it some more, this might not be what we want to do in this situation. As that would require changes in several parts of the configs which I generally try to avoid when not doing a major release.

Another way to do it, which might even make more sense, is to not include it in the routing at root level, but at notification level by adding it to the `urn:solid-server:default:NotificationTypeHandler`. This is also how the Webhook implementation adds the route to the relevent WebID as can be seen here: https://github.com/CommunitySolidServer/CommunitySolidServer/blob/f73dfb31c0fe132524323acf6c4f4636bcd8bc80/config/http/notifications/webhooks/routes.json#L31-L36
The URL would then become something like `/.notifications/StreamingHTTPChannel2023/...` but that actually seems even better to me so everything notification related is in the `/.notifications/` namespace.

The full config of that handler can be seen here: https://github.com/CommunitySolidServer/CommunitySolidServer/blob/f73dfb31c0fe132524323acf6c4f4636bcd8bc80/config/http/notifications/base/http.json#L28-L43



> By the way, after this PR is ready, will I need to make another one for https://github.com/CommunitySolidServer/configuration-generator?

That would be a good idea. I can look into that once this is completely finished though as the changes required there are only minor.

Reply:
Currently, the PR adds the routing to `urn:solid-server:default:NotificationTypeHandler`, but the comments suggest that this is for subscriptions, so we thought to move it elsewhere. If you think it is a good idea to have it there, we will be happy to leave it where it is.

Reply:
Yea having it there is fine. The comment is just there as a guide of what to do for a new notification type. As you can see the webhook solution also puts something else there.